lib/visitor.js

```diff
@@ -6,11 +6,7 @@ const underscoreCode = "_".charCodeAt(0);
 // This Visitor API was inspired by a similar API provided by ast-types:
 // https://github.com/benjamn/ast-types/blob/master/lib/path-visitor.js
 
-function Visitor() {
-  this.visit = this.visit.bind(this);
-  this.visitWithoutReset = this.visitWithoutReset.bind(this);
-  this.visitChildren = this.visitChildren.bind(this);
-}
```

Comment:
Using `bind` is expensive in v8. It was only recently optimized in Node 7. If it can be avoided it should. An arrow function is cheap and this is a parser where performance matters.